src/pages/gen2/build-a-backend/functions/index.mdx

```diff
@@ -59,6 +59,186 @@ defineBackend({
+
+```ts
+// amplify/functions/my-demo-function/handler.ts
+import { env } from '@env/my-demo-function';
+
+export const handler = async (event) => {
+  env.API_KEY; // this is the value of secret named "myApiKey"
+  return 'You made a function!';
+};
+```
+
+## Resource access
+
+When you grant a function access to another resource in your Amplify backend ([such as granting access to storage](../storage/#resource-access)), that will configure environment variables for that function to make SDK calls to the AWS services it has access to. Those environment variables are typed and available as part of the `env` object.
+
+Example defineStorage that grants myDemoFunction access to files in `/foo/*`.
+
+```ts
+// amplify/storage/resource.ts
+import { myDemoFunction } from '../functions/my-demo-function/resource';
+
+export const storage = defineStorage({
+  name: 'myProjectFiles',
+  access: (allow) => ({
+    '/foo/*': [allow.resource(demoFunction).to(['read', 'write', 'delete'])]
+  })
+});
+```
+
+This access definition will add the environment variable `myProjectFiles_BUCKET_NAME` to the function. This environment variable can be accessed on the `env` object.
+
+Here's an example of how it can be used to upload some content to S3.
+
+```ts
+// amplify/functions/my-demo-function/handler.ts
+import { env } from '@env/my-demo-function';
+import { S3Client, PutObjectCommand } from '@aws-sdk/client-s3';
+
+const s3Client = new S3Client();
+
+export const handler = async (event) => {
+  await s3Client.send(
+    new PutObjectCommand({
+      Bucket: env.myProjectFiles_BUCKET_NAME,
+      Key: 'foo/someFile.txt',
+      Body: 'this is an example'
+    })
+  );
+};
+```
+
+## Additional configuration
+
+`defineFunction` comes out-of-the-box with sensible but minimal defaults. The following options are provided to tweak the function configuration.
+
+### `timeoutSeconds`
+
+By default, functions will time out after 3 seconds. This can be configured to any number of seconds up to 15 minutes.
+
+```ts
+// amplify/functions/my-demo-function/resource.ts
+export const myDemoFunction = defineFunction({
+  timeoutSeconds: 60 // 1 minute timeout
+});
+```
+
+### `memoryMB`
+
+By default, functions have 128 MB of memory allocated to them. This can be configured up to 10240 MB. Note that this can increase the cost of executing the function. For more pricing information see [here](https://aws.amazon.com/lambda/pricing/).
+
+```ts
+// amplify/functions/my-demo-function/resource.ts
+export const myDemoFunction = defineFunction({
+  memoryMB: 512 // allocate 512 MB of memory to the function.
+});
+```
+
+### `runtime`
+
+Currently, only Node runtimes are supported by `defineFunction`. However, you can change the Node version that is used by the function. The default is the latest Node LTS version that is supported by AWS Lambda (currently Node 20).
```

Comment:
Node 18 is the current default until a month before its EOL when we are expecting to change the default to 20

Reply:
ah right, it's the oldest LTS that's the default. Good catch! Will update